cmd/promtool/main.go

```diff
@@ -303,9 +304,30 @@ func checkRules(filename string) (int, []error) {
 		numRules += len(rg.Rules)
 	}
 
+	duplicateLabels := checkDuplicateLabels(rgs.Groups)
+	if duplicateLabels != 0 {
+		fmt.Printf("%d duplicate label(s) found. Might cause inconsistency while recording expressions.\n", duplicateLabels)
```

Comment:
It'd be more useful to print the relevant labels. You should also check the metric name.

Reply:
Sure. I will implement it in the following commit.